apps/backend/src/routes/socket/index.ts

```diff
@@ -0,0 +1,120 @@
+import { Server, Socket } from "socket.io";
+import { server } from "@/index";
+import {
+  SocketRequestType,
+  SocketRequestSchema,
+  SocketRequest,
+  SocketResponseType,
+  SocketResponse,
+  SocketErrorPayload,
+  MapRequestToResponse,
+  errorToString
+} from "@types";
+import { Controller } from "@/lib/controller";
+
+export const wsServer = new Server(server);
+
+const controller = new Controller();
+
+const clientsSockets: Record<string, Socket> = {};
+
+const handleError = (socket: Socket, recipient: string, error: string): void => {
+  const payload: SocketErrorPayload = {
+    error,
+  }
+
+  const resp: SocketResponse = {
+    type: SocketResponseType.ERROR,
+    recipientSigPubKey: recipient,
+    senderSigPubKey: "",
+    payload
+  };
+
+  socket.send(JSON.stringify(resp))
+  return;
+}
+
+wsServer.on('connection', (socket: Socket) => {
+  socket.on('open', () => {
+    console.log("Open socket client connection.");
+  });
+
+  socket.on('disconnect', () => {
+    console.log(`Disconnect socket client connection.`);
+  });
+
+  socket.on('message', async (message: string) => {
+    let sender: string | null = null;
+    let recipient: string | null = null;
+    try {
+      const req: SocketRequest = SocketRequestSchema.parse(JSON.parse(message));
+
+      if (!req.senderSigPubKey) {
+        throw new Error("Missing sender");
+      }
+
+      sender = req.senderSigPubKey;
+      recipient = req.recipientSigPubKey;
+
+      if (!req.authToken) {
+        return handleError(socket, sender, "Missing auth token.");
+      }
+
+      // Fetch user by auth token, ensure the request is from an authenticated user
```

Comment:
Not urgent, but if we're doing this verification then the user does not need to send their `senderSigPubKey` with the request

Reply:
https://github.com/cursive-team/connections/pull/191#discussion_r1835542731